detection_rules/packaging.py

```diff
@@ -97,6 +97,9 @@ def __init__(self, rules: RuleCollection, name: str, release: Optional[bool] = F
         if max_version is not None:
             self.rules = self.rules.filter(lambda r: max_version >= r.contents.latest_version)
 
+        if RULES_CONFIG.bypass_version_lock:
```

Comment:
I think we will need to think through this one more (probably in a separate PR).
* what will the concept of a package be _without_ the version lock
* how or _do_ we consider an option for differentiating new, changed, and deprecated rules _without_ a version lock
* perhaps there is minimized functionality that is useful where we can create a separate (inheratance-based) `mini_package` (or better name)`
* and I think several other considerations


For now, I think we should probably advise against package building in the docs and add a TODO to remove this once we finalize the decision

Reply:
assert not RULES_config.bypass_version_lock, "This is a version locking strategy concept. Can not bypass for packaging."

TODO: Create issue.

Reply:
tracker: https://github.com/elastic/detection-rules/issues/3696